book/sending_emails.rst

```diff
@@ -0,0 +1,226 @@
+
+    // src/Acme/DemoBundle/Controller/EmailController.php
+    namespace Acme\DemoBundle\Controller;
+
+    // ...
+
+    class EmailController extends Controller
+    {
+        public function sendMailAction()
+        {
+            // ...
+            $email = ...;
+
+            $processor = $this->get('oro_email.mailer.processor');
+            $processor->process($email);
+        }
+    }
+
+When calling the :method:`Oro\\Bundle\\EmailBundle\\Mailer\\Processor::process` the mailer
+processor performs the following steps:
+
+#. It creates a new ``\Swift_Message`` instance and populate it with the data from your ``Email``
+   object.
+
+#. If you did not pass an ``EmailOrigin`` object which should be used to associate the mail in the
+   user interface with, the processor will create on based on the sender address and the selected
+   organization.
+
+#. The e-mail will be sent based on your application's `SwiftMailer configuration`_.
+
+#. The sent e-mail is persisted to the database storing all necessary information to be able to
+   view it again in the future through the user interface.
+
+#. All the persisted data is returned as an instance of the
+   :class:`Oro\\EmailBundle\\Entity\\EmailUser`.
+
+E-Mail Notifications
+--------------------
+
+Sometimes you want to receive e-mails when entities of a particular class are written to the
+database. To achieve thisthe OroPlatform comes with the NotificationBundle. This bundle registers
+an event listener that is executed whenever a Doctrine entity is created, updated or removed.
+
+To be notified by such an event, you have to create an
+:class:`Oro\\Bundle\\NotificationBundle\\Entity\\EmailNotification` that contains all the necessary
+information. The easiest way to register a new `EmailNotification` is to create data fixtures:
+
+.. code-block:: php
+    :linenos:
+
+    // src/Acme/DemoBundle/Migrations/Data/ORM/CreateCommentNotification.php
+    namespace Acme\DemoBundle\Migrations\Data\ORM;
+
+    use Doctrine\Common\DataFixtures\AbstractFixture;
+    use Doctrine\Common\Persistence\ObjectManager;
+    Oro\Bundle\EmailBundle\Entity\EmailTemplate;
+    use Oro\Bundle\NotificationBundle\Entity\EmailNotification;
+    use Oro\Bundle\NotificationBundle\Entity\Event;
+    use Oro\Bundle\NotificationBundle\Entity\RecipientList;
+    use Oro\Bundle\UserBundle\Entity\Group;
+
+    class CreateCommentNotification extends AbstractFixture
+    {
+        public function load(ObjectManager $manager)
+        {
+            $notification = new EmailNotification();
+
+            // the FQCN of the entity
+            $notification->setEntityName('Acme\DemoBundle\Entity\Comment');
+
+            // the event to be notified of, pre-defined event names are
+            // oro.notification.event.entity_post_update, oro.notification.event.entity_post_remove
+            // and oro.notification.event.entity_post_persist
+            $eventRepository = $manager->getRepository(Event::class);
```

Comment:
Minimal required php version for Oro is 5.4 (http://www.orocrm.com/documentation/index/current/system-requirements)
But ::class began to use since php5.5. So I think you can change this in fixture sample to consistent with system requirements.